src/api/procedures/consumeAuthorizationRequests.ts

```diff
@@ -122,11 +123,13 @@ export async function getAuthorization(
       const account = context.getCurrentAccount();
       condition = target.isEqual(account);
     } else {
-      condition = target.isEqual(identity);
+      did = await fetchDid();
+      condition = did === target.did;
     }
 
     if (!accept) {
-      condition = condition || issuer.isEqual(identity);
+      did = await fetchDid();
+      condition = condition || did === issuer.did;
```

Comment:
I meant you should adapt the optimization to using `isEqual`. Probably the easiest way to go about it is to do something like

```
let identity: Identity;
const fetchIdentity = async () => identity || context.getCurrentIdentity();
```

And use that in a similar manner